pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -284,16 +287,27 @@ func (builder *ListenerBuilder) buildVirtualOutboundListener(
 	}
 
 	actualWildcard, _ := getActualWildcardAndLocalHost(node)
+	var listFilter []*listener.ListenerFilter
+	if util.IsAllowAnyOutbound(node) && node.SidecarScope.OutboundTrafficPolicy.EgressProxy != nil {
+		listFilter = append(listFilter, &listener.ListenerFilter{
+			Name: xdsutil.TlsInspector,
+		})
+	}
 
 	// add an extra listener that binds to the port that is the recipient of the iptables redirect
 	ipTablesListener := &xdsapi.Listener{
 		Name:             VirtualOutboundListenerName,
 		Address:          util.BuildAddress(actualWildcard, uint32(push.Mesh.ProxyListenPort)),
 		Transparent:      isTransparentProxy,
 		UseOriginalDst:   proto.BoolTrue,
+		ListenerFilters:  listFilter,
 		FilterChains:     filterChains,
 		TrafficDirection: core.TrafficDirection_OUTBOUND,
 	}
+	ipTablesListener.ListenerFiltersTimeout = gogo.DurationToProtoDuration(push.Mesh.ProtocolDetectionTimeout)
```

Comment:
do we need to only set this if we have a tls inspector or does it not matter?

Reply:
it doesnt matter if there are no filters. This will become required once I add the http sniffing for the virtual outbound